File: TriangleMaking/solve/cpp/TriangleMakingSolver.cpp
#include <fstream>
#include <topcoder>
#include "TriangleMaking.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int a; tc::read(ifs, a); tc::next(ifs);
    int b; tc::read(ifs, b); tc::next(ifs);
    int c; tc::read(ifs, c);
    ifs.close();

    std::ofstream ofs(argv[2]);
    TriangleMaking solver;
    tc::show(ofs, solver.maxPerimeter(a, b, c));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: TriangleMaking/solve/java/TriangleMakingSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class TriangleMakingSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        int a = (Integer) reader.next(Integer.class);
        reader.next();
        
        int b = (Integer) reader.next(Integer.class);
        reader.next();
        
        int c = (Integer) reader.next(Integer.class);
        reader.close();

        TriangleMaking solver = new TriangleMaking();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.maxPerimeter(a, b, c));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}
